From 76681895c15d87b19af867b41504c3484bf62eae Mon Sep 17 00:00:00 2001
From: Jin Li <jinl@codeaurora.org>
Date: Tue, 31 May 2016 10:47:45 -0400
Subject: [PATCH 41/44] defconfig: enable drm/kms driver

Change to use DRM/KMS driver and disable FB driver.

Change-Id: I7691cd1d35d09ae2d858c0f6aff100ba194e4258
Signed-off-by: Jin Li <jinl@codeaurora.org>
---
 arch/arm64/configs/msm-auto-perf_defconfig | 77 +++++++++++++++++++++++-------
 arch/arm64/configs/msm-auto_defconfig      | 12 ++---
 2 files changed, 63 insertions(+), 26 deletions(-)

diff --git a/arch/arm64/configs/msm-auto-perf_defconfig b/arch/arm64/configs/msm-auto-perf_defconfig
index 8411926..b1dc352 100644
--- a/arch/arm64/configs/msm-auto-perf_defconfig
+++ b/arch/arm64/configs/msm-auto-perf_defconfig
@@ -1,4 +1,3 @@
-CONFIG_LOCALVERSION="-perf"
 CONFIG_AUDIT=y
 # CONFIG_AUDITSYSCALL is not set
 CONFIG_NO_HZ=y
@@ -9,6 +8,7 @@ CONFIG_IKCONFIG=y
 CONFIG_IKCONFIG_PROC=y
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=15
 CONFIG_CGROUPS=y
+CONFIG_CGROUP_DEBUG=y
 CONFIG_CGROUP_FREEZER=y
 CONFIG_CGROUP_CPUACCT=y
 CONFIG_RESOURCE_COUNTERS=y
@@ -18,14 +18,12 @@ CONFIG_SCHED_HMP=y
 CONFIG_NAMESPACES=y
 # CONFIG_UTS_NS is not set
 # CONFIG_PID_NS is not set
-CONFIG_RELAY=y
 CONFIG_BLK_DEV_INITRD=y
 CONFIG_RD_BZIP2=y
 CONFIG_RD_LZMA=y
 CONFIG_CC_OPTIMIZE_FOR_SIZE=y
 CONFIG_KALLSYMS_ALL=y
 CONFIG_EMBEDDED=y
-# CONFIG_SLUB_DEBUG is not set
 CONFIG_PROFILING=y
 CONFIG_CC_STACKPROTECTOR_REGULAR=y
 CONFIG_MODULES=y
@@ -38,6 +36,7 @@ CONFIG_MODULE_SIG_SHA512=y
 CONFIG_PARTITION_ADVANCED=y
 CONFIG_ARCH_MSM=y
 CONFIG_ARCH_MSM8996=y
+CONFIG_ARCH_MSMCOBALT=y
 CONFIG_PCI_MSM=y
 CONFIG_ENABLE_FP_SIMD_SETTINGS=y
 CONFIG_SMP=y
@@ -59,6 +58,8 @@ CONFIG_PM_WAKELOCKS=y
 CONFIG_PM_WAKELOCKS_LIMIT=0
 # CONFIG_PM_WAKELOCKS_GC is not set
 CONFIG_PM_RUNTIME=y
+CONFIG_PM_DEBUG=y
+CONFIG_SUSPEND_TIME=y
 CONFIG_CPU_IDLE=y
 CONFIG_CPU_FREQ=y
 CONFIG_SCHED_FREQ_INPUT=y
@@ -310,6 +311,8 @@ CONFIG_INPUT_GPIO=y
 # CONFIG_DEVMEM is not set
 # CONFIG_DEVKMEM is not set
 CONFIG_SERIAL_MSM_HS=y
+CONFIG_SERIAL_MSM_HSL=y
+CONFIG_SERIAL_MSM_HSL_CONSOLE=y
 CONFIG_SERIAL_MSM_SMD=y
 CONFIG_DIAG_CHAR=y
 CONFIG_HW_RANDOM=y
@@ -352,7 +355,6 @@ CONFIG_THERMAL_QPNP=y
 CONFIG_THERMAL_QPNP_ADC_TM=y
 CONFIG_WCD9330_CODEC=y
 CONFIG_WCD9335_CODEC=y
-CONFIG_REGULATOR=y
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
 CONFIG_REGULATOR_PROXY_CONSUMER=y
 CONFIG_REGULATOR_STUB=y
@@ -408,15 +410,9 @@ CONFIG_RADIO_SILABS=y
 # CONFIG_MEDIA_SUBDRV_AUTOSELECT is not set
 CONFIG_VIDEO_ADV7481=y
 CONFIG_MSM_KGSL=y
-CONFIG_FB=y
-CONFIG_FB_MSM=y
-CONFIG_MSM_BA_V4L2=y
+CONFIG_DRM=y
 CONFIG_MSM_DBA=y
 CONFIG_MSM_DBA_ADV7533=y
-CONFIG_FB_MSM_MDSS=y
-CONFIG_FB_MSM_MDSS_WRITEBACK=y
-CONFIG_FB_MSM_MDSS_HDMI_PANEL=y
-CONFIG_FB_MSM_MDSS_XLOG_DEBUG=y
 CONFIG_BACKLIGHT_LCD_SUPPORT=y
 CONFIG_SOUND=y
 CONFIG_SND=y
@@ -479,6 +475,7 @@ CONFIG_RTC_DRV_QPNP=y
 CONFIG_ESOC=y
 CONFIG_ESOC_DEV=y
 CONFIG_ESOC_CLIENT=y
+CONFIG_ESOC_DEBUG=y
 CONFIG_ESOC_MDM_4x=y
 CONFIG_ESOC_MDM_DRV=y
 CONFIG_DMADEVICES=y
@@ -491,12 +488,15 @@ CONFIG_ANDROID_BINDER_IPC=y
 CONFIG_ASHMEM=y
 CONFIG_ANDROID_TIMED_GPIO=y
 CONFIG_ANDROID_LOW_MEMORY_KILLER=y
+CONFIG_SYNC=y
+CONFIG_SW_SYNC=y
 CONFIG_SW_SYNC_USER=y
 CONFIG_ION=y
 CONFIG_ION_MSM=y
 CONFIG_MSM_AVTIMER=y
 CONFIG_MSM_BUS_SCALING=y
 CONFIG_BUS_TOPOLOGY_ADHOC=y
+CONFIG_DEBUG_BUS_VOTER=y
 CONFIG_QPNP_POWER_ON=y
 CONFIG_QPNP_REVID=y
 CONFIG_QPNP_COINCELL=y
@@ -516,6 +516,9 @@ CONFIG_MSM_ADSP_LOADER=y
 CONFIG_MSM_MEMORY_DUMP_V2=y
 CONFIG_MSM_BOOT_STATS=y
 CONFIG_MSM_BOOT_TIME_MARKER=y
+CONFIG_MSM_CPUSS_DUMP=y
+CONFIG_MSM_COMMON_LOG=y
+CONFIG_MSM_DDR_HEALTH=y
 CONFIG_MSM_WATCHDOG_V2=y
 CONFIG_MSM_FORCE_WDOG_BITE_ON_PANIC=y
 CONFIG_MSM_CACHE_M4M_ERP64=y
@@ -541,6 +544,7 @@ CONFIG_MSM_SMP2P_TEST=y
 CONFIG_MSM_SPM=y
 CONFIG_MSM_L2_SPM=y
 CONFIG_MSM_QMI_INTERFACE=y
+CONFIG_MSM_DCC=y
 CONFIG_MSM_EVENT_TIMER=y
 CONFIG_MSM_SYSMON_GLINK_COMM=y
 CONFIG_MSM_IPC_ROUTER_GLINK_XPRT=y
@@ -552,6 +556,8 @@ CONFIG_MSM_PIL_MSS_QDSP6V5=y
 CONFIG_TRACER_PKT=y
 CONFIG_MSM_CORE_CTL_HELPER=y
 CONFIG_MSM_PERFORMANCE=y
+CONFIG_MSM_SERVICE_LOCATOR=y
+CONFIG_MSM_SERVICE_NOTIFIER=y
 CONFIG_MSM_QBT1000=y
 CONFIG_MSM_KERNEL_PROTECT=y
 CONFIG_MEM_SHARE_QMI_SERVICE=y
@@ -560,17 +566,18 @@ CONFIG_MSM_MDSS_PLL=y
 CONFIG_REMOTE_SPINLOCK_MSM=y
 CONFIG_ARM_SMMU=y
 CONFIG_IOMMU_DEBUG=y
+CONFIG_IOMMU_DEBUG_TRACKING=y
 CONFIG_IOMMU_TESTS=y
 CONFIG_SPDM_SCM=y
 CONFIG_DEVFREQ_SPDM=y
 CONFIG_PWM=y
 CONFIG_PWM_QPNP=y
+CONFIG_ARM_GIC_PANIC_HANDLER=y
 CONFIG_ARM_GIC_V3_ACL=y
 CONFIG_CORESIGHT=y
 CONFIG_CORESIGHT_EVENT=y
 CONFIG_CORESIGHT_FUSE=y
 CONFIG_CORESIGHT_CTI=y
-CONFIG_CORESIGHT_CTI_SAVE_DISABLE=y
 CONFIG_CORESIGHT_TMC=y
 CONFIG_CORESIGHT_TPIU=y
 CONFIG_CORESIGHT_FUNNEL=y
@@ -579,9 +586,12 @@ CONFIG_CORESIGHT_TPDA=y
 CONFIG_CORESIGHT_TPDM=y
 CONFIG_CORESIGHT_STM=y
 CONFIG_CORESIGHT_HWEVENT=y
+CONFIG_CORESIGHT_ETMV4=y
+CONFIG_CORESIGHT_REMOTE_ETM=y
 CONFIG_CORESIGHT_QPDI=y
 CONFIG_SENSORS_SSC=y
 CONFIG_MSM_TZ_LOG=y
+CONFIG_MOBICORE_DRIVER=m
 CONFIG_EXT2_FS=y
 CONFIG_EXT2_FS_XATTR=y
 CONFIG_EXT3_FS=y
@@ -593,7 +603,6 @@ CONFIG_QUOTA=y
 CONFIG_FUSE_FS=y
 CONFIG_MSDOS_FS=y
 CONFIG_VFAT_FS=y
-CONFIG_TMPFS=y
 CONFIG_TMPFS_POSIX_ACL=y
 CONFIG_ECRYPT_FS=y
 CONFIG_ECRYPT_FS_MESSAGING=y
@@ -601,21 +610,55 @@ CONFIG_NLS_CODEPAGE_437=y
 CONFIG_NLS_ASCII=y
 CONFIG_NLS_ISO8859_1=y
 CONFIG_PRINTK_TIME=y
+CONFIG_LOG_BUF_MAGIC=y
+CONFIG_DYNAMIC_DEBUG=y
 CONFIG_DEBUG_INFO=y
+CONFIG_PAGE_OWNER=y
+CONFIG_MAGIC_SYSRQ=y
+CONFIG_DEBUG_OBJECTS=y
+CONFIG_DEBUG_OBJECTS_FREE=y
+CONFIG_DEBUG_OBJECTS_TIMERS=y
+CONFIG_DEBUG_OBJECTS_WORK=y
+CONFIG_DEBUG_OBJECTS_RCU_HEAD=y
+CONFIG_DEBUG_OBJECTS_PERCPU_COUNTER=y
+CONFIG_DEBUG_KMEMLEAK=y
+CONFIG_DEBUG_KMEMLEAK_DEFAULT_OFF=y
+CONFIG_DEBUG_STACK_USAGE=y
+CONFIG_DEBUG_MEMORY_INIT=y
+CONFIG_LOCKUP_DETECTOR=y
+CONFIG_BOOTPARAM_SOFTLOCKUP_PANIC=y
+# CONFIG_DETECT_HUNG_TASK is not set
 CONFIG_PANIC_TIMEOUT=5
 CONFIG_PANIC_ON_RECURSIVE_FAULT=y
+CONFIG_PANIC_ON_SCHED_BUG=y
+CONFIG_PANIC_ON_RT_THROTTLING=y
 CONFIG_SCHEDSTATS=y
 CONFIG_TIMER_STATS=y
-# CONFIG_DEBUG_PREEMPT is not set
-# CONFIG_DEBUG_BUGVERBOSE is not set
+CONFIG_DEBUG_SPINLOCK=y
+CONFIG_DEBUG_MUTEXES=y
+CONFIG_DEBUG_ATOMIC_SLEEP=y
+CONFIG_DEBUG_LIST=y
+CONFIG_FAULT_INJECTION=y
+CONFIG_FAIL_PAGE_ALLOC=y
+CONFIG_UFS_FAULT_INJECTION=y
+CONFIG_FAULT_INJECTION_DEBUG_FS=y
+CONFIG_FAULT_INJECTION_STACKTRACE_FILTER=y
+CONFIG_MSM_RTB=y
+CONFIG_MSM_RTB_SEPARATE_CPUS=y
 CONFIG_IPC_LOGGING=y
+CONFIG_IRQSOFF_TRACER=y
+CONFIG_PREEMPT_TRACER=y
+CONFIG_BLK_DEV_IO_TRACE=y
 CONFIG_CPU_FREQ_SWITCH_PROFILER=y
+CONFIG_PANIC_ON_DATA_CORRUPTION=y
+CONFIG_ARM64_PTDUMP=y
 CONFIG_DEBUG_SET_MODULE_RONX=y
-CONFIG_DEBUG_RODATA=y
-CONFIG_DEBUG_ALIGN_RODATA=y
+CONFIG_FREE_PAGES_RDONLY=y
+CONFIG_KERNEL_TEXT_RDONLY=y
 CONFIG_PFK=y
 CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
+CONFIG_LSM_MMAP_MIN_ADDR=4096
 CONFIG_SECURITY_SELINUX=y
 # CONFIG_INTEGRITY is not set
 CONFIG_CRYPTO_NULL=y
diff --git a/arch/arm64/configs/msm-auto_defconfig b/arch/arm64/configs/msm-auto_defconfig
index 9a1856e..b1dc352 100644
--- a/arch/arm64/configs/msm-auto_defconfig
+++ b/arch/arm64/configs/msm-auto_defconfig
@@ -355,7 +355,6 @@ CONFIG_THERMAL_QPNP=y
 CONFIG_THERMAL_QPNP_ADC_TM=y
 CONFIG_WCD9330_CODEC=y
 CONFIG_WCD9335_CODEC=y
-CONFIG_REGULATOR=y
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
 CONFIG_REGULATOR_PROXY_CONSUMER=y
 CONFIG_REGULATOR_STUB=y
@@ -411,15 +410,9 @@ CONFIG_RADIO_SILABS=y
 # CONFIG_MEDIA_SUBDRV_AUTOSELECT is not set
 CONFIG_VIDEO_ADV7481=y
 CONFIG_MSM_KGSL=y
-CONFIG_FB=y
-CONFIG_FB_MSM=y
-CONFIG_MSM_BA_V4L2=y
+CONFIG_DRM=y
 CONFIG_MSM_DBA=y
 CONFIG_MSM_DBA_ADV7533=y
-CONFIG_FB_MSM_MDSS=y
-CONFIG_FB_MSM_MDSS_WRITEBACK=y
-CONFIG_FB_MSM_MDSS_HDMI_PANEL=y
-CONFIG_FB_MSM_MDSS_XLOG_DEBUG=y
 CONFIG_BACKLIGHT_LCD_SUPPORT=y
 CONFIG_SOUND=y
 CONFIG_SND=y
@@ -495,6 +488,8 @@ CONFIG_ANDROID_BINDER_IPC=y
 CONFIG_ASHMEM=y
 CONFIG_ANDROID_TIMED_GPIO=y
 CONFIG_ANDROID_LOW_MEMORY_KILLER=y
+CONFIG_SYNC=y
+CONFIG_SW_SYNC=y
 CONFIG_SW_SYNC_USER=y
 CONFIG_ION=y
 CONFIG_ION_MSM=y
@@ -608,7 +603,6 @@ CONFIG_QUOTA=y
 CONFIG_FUSE_FS=y
 CONFIG_MSDOS_FS=y
 CONFIG_VFAT_FS=y
-CONFIG_TMPFS=y
 CONFIG_TMPFS_POSIX_ACL=y
 CONFIG_ECRYPT_FS=y
 CONFIG_ECRYPT_FS_MESSAGING=y
-- 
2.8.2

